index.js

```diff
@@ -0,0 +1,69 @@
+export default function createMap(equals) {
+	const map = new Array(256)
+	return {
+		get: (key) => lookup(equals, map, key, 0),
+		set: (key, value) => set(equals, map, key, 0, value),
+		delete: (key) => remove(equals, map, key, 0),
+		getReference: (key, insert = false) =>
+			getReference(equals, map, key, 0, insert),
+	}
+}
+
+function set(equals, map, key, keyOffset, value) {
+	const k = key[keyOffset]
+	const r = map[k]
```

Comment:
What about doing some validation here?
```
	if(!(map instanceof Uint8Array)) throw TypeError('Uint8Array expected')
	if(!(key instanceof Uint8Array)) throw TypeError('Uint8Array expected')
	if (keyOffset >= key.length) throw new Error('Prefixed key')
```

Reply:
If it's zero-cost to performance, I could be in for type-checking. I'm considering though if the long-term strategy should be TypeScript then.